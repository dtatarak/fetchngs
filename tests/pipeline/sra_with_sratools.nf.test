nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    profile "test,docker"
    tag "pipeline"

    test("Force SRA-Tools Download") {

        when {
            params {
                outdir                  = "$outputDir"
                force_sratools_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("metadata") },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }

    }

}
