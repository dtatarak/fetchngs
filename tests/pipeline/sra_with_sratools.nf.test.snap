{
    "metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,f46819ed848d0d28dff80fd0f5a4cd34",
            "DRR028935.runinfo_ftp.tsv:md5,ef08c3756034e6705b0e13d1f37bfea5",
            "ERR1109373.runinfo_ftp.tsv:md5,6e7960f43046c6080435a024e9c426c9",
            "ERR1160846.runinfo_ftp.tsv:md5,db8b7bc5877a62b90e719d7cf3c0d25f",
            "SRR12848126.runinfo_ftp.tsv:md5,b8ac151eb85a001e9aa3d7319156c016",
            "SRR13191702.runinfo_ftp.tsv:md5,7188ac3b1d73ec297a900450e73bf751",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,827751e211294eeecbb7239f70ead9a5"
        ],
        "timestamp": "2023-05-09T02:12:04+0000"
    },
    "samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d",
            "samplesheet.csv:md5,24d51c3c83f25902d1841096b12e2d67"
        ],
        "timestamp": "2023-05-09T02:12:04+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,ec1f5ee8c99879c3b8f04e7e683f540f"
        ],
        "timestamp": "2023-05-09T02:12:04+0000"
    },
    "fastq": {
        "content": [
            "DRX024467_DRR026872.fastq.gz:md5,bd3f113c3b06cf4499fa0e00c634ec85",
            "DRX026011_DRR028935_1.fastq.gz:md5,0f96fb60e903c8cbd55a6977e66f7c93",
            "DRX026011_DRR028935_2.fastq.gz:md5,b97b1e0e806bcae10a7443e2f22d6ca2",
            "ERX1188904_ERR1109373_1.fastq.gz:md5,e7edae2715dfaf4c318d37959fd9ae22",
            "ERX1188904_ERR1109373_2.fastq.gz:md5,34ab048c92201a5ec4ab36d8dc1c5732",
            "ERX1234253_ERR1160846.fastq.gz:md5,22683b24b2467494e430643423f1e275",
            "SRX10940790_SRR14593545_1.fastq.gz:md5,6a4762cf0f1f9f5544d3735e2e07c5d3",
            "SRX10940790_SRR14593545_2.fastq.gz:md5,cbd30715fdef457d7aa24a12f19eedf2",
            "SRX11047067_SRR14709033.fastq.gz:md5,79140dd1cb3fd1c5e02d88e01f7c3eda",
            "SRX6725035_SRR9984183.fastq.gz:md5,7a70e678c400366869601c8be41fe74b",
            "SRX9315476_SRR12848126_1.fastq.gz:md5,03f5a34586aef9cd61f54eb26831a665",
            "SRX9315476_SRR12848126_2.fastq.gz:md5,97701c5d884af8d96ba95d7237f3bf79",
            "SRX9626017_SRR13191702_1.fastq.gz:md5,627a90afcd9f3cfc7f5f91074f9a2df7",
            "SRX9626017_SRR13191702_2.fastq.gz:md5,3d9b2c68a8ad1547105b17de81b2d9af"
        ],
        "timestamp": "2023-05-09T02:12:04+0000"
    }
}